frontend/tests/utils/date-utils.test.ts

```diff
@@ -0,0 +1,23 @@
+import { describe, expect, it } from 'vitest';
+
+import { isValidTimeZone } from '~/utils/date-utils';
+
+describe('date-utils', () => {
+  describe('isValidTimeZone', () => {
+    it('should return true for valid timezones', () => {
+      expect(isValidTimeZone('UTC')).toEqual(true);
+      expect(isValidTimeZone('Canada/Newfoundland')).toEqual(true);
+      expect(isValidTimeZone('Canada/Atlantic')).toEqual(true);
+      expect(isValidTimeZone('Canada/Central')).toEqual(true);
+      expect(isValidTimeZone('Canada/Mountain')).toEqual(true);
+      expect(isValidTimeZone('Canada/Pacific')).toEqual(true);
+    });
+
+    it('should return false for invalid timezones', () => {
+      expect(isValidTimeZone('')).toEqual(false);
+      expect(isValidTimeZone('Canada')).toEqual(false);
+      expect(isValidTimeZone('MyTimezone!')).toEqual(false);
+      expect(isValidTimeZone('Canada/Los_Angeles')).toEqual(false);
+    });
+  });
```

Comment:
Use parameterized tests; otherwise, a failure in one assertion could prevent you from knowing if the subsequent assertions pass.

```suggestion
  describe('isValidTimeZone', () => {
    it.each([
      ['UTC', true],
      ['Canada/Newfoundland', true],
      ['Canada/Atlantic', true], 
      ['Canada/Central', true],
      ['Canada/Mountain', true],
      ['Canada/Pacific', true],
      ['', false],
      ['Canada', false],
      ['MyTimezone!', false],
      ['Canada/Los_Angeles', false],
    ])('should validate timezone "%s" to be %s', (timezone, expected) => {
      expect(isValidTimeZone(timezone)).toEqual(expected);
    });
  });
```

Reply:
Did it a little differently: e111c1d1